dumpling/export/dump.go

```diff
@@ -331,11 +333,17 @@ func (d *Dumper) dumpDatabases(tctx *tcontext.Context, metaConn *sql.Conn, taskC
 		}
 	}
 
+	parser1 := parser.New()
 	for dbName, tables := range allTables {
 		if !conf.NoSchemas {
 			createDatabaseSQL, err := ShowCreateDatabase(metaConn, dbName)
 			if err != nil {
-				return err
+				return errors.Trace(err)
+			}
+			// adjust db collation
+			createDatabaseSQL, err = adjustDatabaseCollation(tctx, metaConn, parser1, createDatabaseSQL, dbName)
```

Comment:
Can we get all database collations in one sql and save them at the beginning of this function and save them in one map?